### PR TITLE
Server-side tracking options

### PR DESCRIPTION
**Pull request by jkao**
This is branched off the 3.2 branch.

I don't think we'll make any of these diffs until that lands, but figured it may be worth a look for now. We'll need to test with either:

1) an example app
2) toolkit

...or I suppose both.

### CODE REVIEW

File: RadarSDK/Radar.h [hunk 1]
```diff
@@ -382,6 +382,13 @@ typedef void (^_Nonnull RadarRouteMatrixCompletionHandler)(RadarStatus status, R
  */
 + (void)startTrackingWithOptions:(RadarTrackingOptions *)options NS_SWIFT_NAME(startTracking(trackingOptions:));
 
+/**
+ Starts tracking the user's location in the background with remotely configured tracking options.
+
+ @see https://radar.io/documentation/sdk/ios#background-tracking-for-geofencing
```
Comment (jkao) [inline RadarSDK/Radar.h:388]: We'll probably need some new docs, but I kept this the same as above.

File: RadarSDK/RadarAPIClient.m [hunk 2]
```diff
@@ -58,7 +58,30 @@ + (NSDictionary *)headersWithPublishableKey:(NSString *)publishableKey {
...
++ (void)parseMetaAndConfigure:(NSDictionary *_Nullable)res {
+    if (!res) {
+        return;
+    }
+
+    id metaObj = res[@"meta"];
+    if (metaObj && [metaObj isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *meta = (NSDictionary *)metaObj;
+        id configObj = meta[@"config"];
+        if (configObj && [configObj isKindOfClass:[NSDictionary class]]) {
```
Comment (jkao) [inline RadarSDK/RadarAPIClient.m:70]: I prefer the go-style of `return`-ing early, but this was just an extraction of existing logic. Will probably do that here.

File: RadarSDK/RadarAPIClient.m [hunk 3]
```diff
@@ -177,13 +192,15 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
...
     }
+    
     RadarTrackingOptions *options = [RadarSettings trackingOptions];
     if (options.syncGeofences) {
         params[@"nearbyGeofences"] = @(YES);
     }
     if (nearbyBeacons) {
         params[@"nearbyBeacons"] = nearbyBeacons;
     }
+    params[@"trackingOptions"] = [options dictionaryValue];
```
Comment (jkao) [inline RadarSDK/RadarAPIClient.m:203]: This is sent up and serialized to our locations cold storage for debugging.

File: RadarSDK/Radar.m [hunk 4]
```diff
@@ -186,9 +186,17 @@ + (void)trackOnceWithLocation:(CLLocation *)location completionHandler:(RadarTra
...
 
 + (void)startTrackingWithOptions:(RadarTrackingOptions *)options {
+    [RadarSettings setListenToServerTrackingOptions:NO];
     [[RadarLocationManager sharedInstance] startTrackingWithOptions:options];
 }
 
++ (void)startTrackingWithRemoteOptions {
+    [RadarSettings setListenToServerTrackingOptions:YES];
+    [[RadarAPIClient sharedInstance] getConfig:^(RadarStatus status, NSDictionary *_Nullable res) {
+        [[RadarLocationManager sharedInstance] startTrackingWithOptions:[RadarSettings trackingOptions]];
```
Comment (jsani-radar) [inline RadarSDK/Radar.m:196]: on first read I expected `res` to be passed through the function as the options arg - it feels like a bit of misdirection that we're setting the storing the config while parsing the meta and then refetching the config here?

File: RadarSDK/RadarAPIClient.m [hunk 5]
```diff
@@ -58,7 +58,30 @@ + (NSDictionary *)headersWithPublishableKey:(NSString *)publishableKey {
     };
 }
 
-- (void)getConfig {
++ (void)parseMetaAndConfigure:(NSDictionary *_Nullable)res {
```
Comment (jsani-radar) [inline RadarSDK/RadarAPIClient.m:61]: `parseMetaAndSetConfig`?

File: RadarSDK/RadarAPIClient.m (hunk 3 above)
Comment (jsani-radar) [reply RadarSDK/RadarAPIClient.m:203]: add as comment?

**Review by jsani-radar — COMMENTED** [review]
few small comments, overall looking solid

**Pull request merged**